File: Data Structures & Algorithms/max-water-container/submission-0.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let max = 0;

        let left = 0;
        let right = heights.length - 1;

        while(left < right){
            let smaller = Math.min(heights[left], heights[right]);
            let area = smaller * (right - left);
            max = Math.max(area, max);

            if(heights[left] > heights[right]){
                right--;
            }else{
                left++;
            }
        }

        return max;
    }
}

File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def lowestCommonAncestor(self, root: TreeNode, p: TreeNode, q: TreeNode) -> TreeNode:
        def search(node):
            if not node or not p or not q:
                return None
            if max(p.val, q.val) < node.val:
                return search(node.left)
            elif min(p.val, q.val) > node.val:
                return search(node.right)
            else:
                return node

        
        return search(root)